File: application/backend-credit-card-enrollment/backend-java/src/main/java/cloud/ambar/common/ambar/AmbarResponseFactory.java
package cloud.ambar.common.ambar;

public class AmbarResponseFactory {
    public static String retryResponse(Exception exception) {
        String message = exception.getMessage() != null ? exception.getMessage() : "";
        message = message.replace("\"", "\\\"");
        return "{\"result\":{\"error\":{\"policy\":\"must_retry\",\"class\":\"" + exception.getClass() + "\",\"description\":\"message:" + message + "\"}}}";
    }

    public static String successResponse() {
        return "{\"result\":{\"success\":{}}}";
    }
}

File: application/backend-credit-card-enrollment/backend-c#/Common/Ambar/AmbarResponseFactory.cs
namespace CreditCardEnrollment.Common.Ambar;

public static class AmbarResponseFactory
{
    public static string RetryResponse(Exception exception)
    {
        var message = exception.Message.Replace("\"", "\\\"");
        return $"{{\"result\":{{\"error\":{{\"policy\":\"must_retry\",\"class\":\"{exception.GetType()}\",\"description\":\"message:{message}\"}}}}}}";
    }

    public static string SuccessResponse()
    {
        return "{\"result\":{\"success\":{}}}";
    }
}